term_proposal/Term_proposal.md

```diff
@@ -0,0 +1,27 @@
+# XGBoost implementation.
+
+I am Alisher, PhD student, EECS International Graduate Program, National Chiao Tung University.
+
+## Introduction
+In statistics and machine learning, ensemble methods use multiple learning algorithms to obtain better predictive performance than could be obtained from any of the constituent learning algorithms alone. Common types of ensembles: Bayes optimal classifier, Bootstrap aggregating (bagging) and Boosting. This project will be focus on Boosting algorithm. Boosting is a machine learning ensemble meta-algorithm for primarily reducing bias, and also variance in supervised learning, and a family of machine learning algorithms that convert weak learners to strong ones. One of the most popular boosting algorithm among the Kaggle community is XGBoost. 
+
+## Why do I want to do this project?
+In my last Kaggle competition related to an image classification problem, the victory was won by a girl from Japan who used XGBoost algorithm. Even this algorithm is very popular, the most people just try to tune hyperparameters for training the model. But very few people implement XGBoost by their own. The reason is obvious, if a powerful enough tool already exists, then why come up with something new.
+However, I think that for education process is really helpful when you don’t use libraries, but try to implement it by yourself. It can give more insight into how XGBoost works.
```

Comment:
I agree implementing a successful tool from ground up will help you a lot for developing skills in computational science.  It's also something you don't always have the luxury to do in a research project.  And it's certainly the purpose of the term project of this course.  I don't know about XGBoost, but if my understanding (from your description) is correct, it's something related to neural network and/or linear algebra with the application of machine learning, and it fits my expectation of a numerical software project.

I'd hope you can be more specific about what you'd like to practice (the educational purpose) in the proceeding sections.  I see you made some good starting points and will leave concrete suggestions accordingly.  Here I'd like you to think about in a global sense.  You should state in the proposal what do you expect to learn from this experience.

Reply:
Dear Yung-Yu. Your understanding is completely correct. The purpose of this project is better understanding of this method. Because now I just know how to apply it, but I don't understand why XGBoost works so well. So I hope that implementing small part of this powerful tool gives me good understanding of this.